appengine/pond_storage/get.py

```diff
@@ -19,18 +19,99 @@
 
 import cgi
 import json
+import logging
 from google.appengine.ext import ndb
+from google.appengine.api import users
 from pond_storage import *
 
+
+def format_top_ducks(top_ducks_info):
+  """Adds an isOwner property to top ducks that are owned by the current user.
+  Args:
+    top_ducks_info: A list of ducks with the best rankings.
+  Returns:
+    A list of dictionaries containing duck information.
+    Example:
+    {
+      'name': 'bob',
+      'duck_key': 'urlsafe duck key',
+      'code': {'js': 'some code', 'opt_xml': 'some xml'},
+      'published': 'true',
+      'isOwner':'True'
+    }
+  """
+  formatted_ducks = []
+  user_key = get_user_key(users.get_current_user())
+  user_ducks = Duck.query(ancestor=user_key).fetch()
+
+  for duck_info in top_ducks_info:
+    matches = [x for x in user_ducks if x.key.urlsafe() == duck_info['duck_key']]
+    if len(matches) > 0:
+      duck_info['isOwner'] = True
+    formatted_ducks.append(duck_info)
+  return formatted_ducks
```

Comment:
I think you should be able to do this check in get_duck_info also without needing to do a query for user_ducks.
Something like this might work:
`isOwner = duck.ancestor == get_user_key(users.get_current_user())`

Reply:
Actually, it should be:
`isOwner = duck.key.parent() == get_user_key(users.get_current_user())`

Reply:
done